ksqldb-execution/src/main/java/io/confluent/ksql/schema/query/QuerySchemas.java

```diff
@@ -129,24 +144,35 @@ public SchemaInfo getTopicInfo(final String topicName) {
 
     final Set<String> keyLoggerNames = kvLoggerNames.getOrDefault(true, ImmutableSet.of());
     if (keyLoggerNames.size() != 1) {
-      throw new IllegalStateException("Multiple key logger names registered for topic."
+      final String result = keyLoggerNames.size() == 0 ? "Zero" : "Multiple";
+
+      throw new IllegalStateException(result + " key logger names registered for topic."
           + System.lineSeparator()
           + "topic: " + topicName
           + "loggers: " + keyLoggerNames
       );
     }
 
     final Set<String> valueTopicNames = kvLoggerNames.getOrDefault(false, ImmutableSet.of());
-    if (valueTopicNames.size() != 1) {
-      throw new IllegalStateException("Multiple value logger names registered for topic."
+    if (valueTopicNames.size() > 1) {
+      LOG.debug("Multiple value logger names registered for topic."
           + System.lineSeparator()
           + "topic: " + topicName
           + "loggers: " + valueTopicNames
       );
     }
 
     final SchemaInfo keyInfo = loggerToSchemas.get(Iterables.getOnlyElement(keyLoggerNames));
-    final SchemaInfo valueInfo = loggerToSchemas.get(Iterables.getOnlyElement(valueTopicNames));
+
+    // If valueTopicsNames has multiple logger names, then takes only the first logger found.
+    // Multiple loggers may be caused by a topic that has multiple value formats, i.e. the shared
+    // outer store from left/outer joins supports multiple value formats. Currently, the
+    // getTopicInfo() method is only called by ksqldb-functional-tests, which uses the same
+    // value format on both left/right joins.
```

Comment:
> which uses the same value format on both left/right joins.

As discussed in person. I this actually correct, ie, can we rely on it? Do we have test for mix value formats for stream-stream joins?

Reply:
I'm not sure if add a test for this because it should be supported. But the QTT is the only client using the `getTopicInfo` and had the issue with the multiple value formats. I added code just to verify multiple values have the same format. I don't think the `getTopicInfo` will be called by anyone else. 

We should refactor the code, though, so it is only used by testing.